### PR TITLE
Added information about deployment to Microsoft IIS

### PR DESCRIPTION
**Pull request by HaraldMuehlhoffCC**
## Description

I ran into caching issues after deploying my first Gatsby project to Microsoft Internet Information Server (IIS). Working with the information on gatsbyjs.org I was able to resolve the issues but it still took some effort.

My [post on twitter](https://twitter.com/HaraldMuehlhoff/status/1191299093265604608) was commented on by  the official gatsbyjs Twitter account "ooo this would be great info to add to our docs!". So here we are!

I'm aware that much more could be said about IIS configuration but the given information should help most users with some experience and at least be a starting point.

### CODE REVIEW

**Comment by muescha** [comment]
it should also here a link to the doc: https://www.gatsbyjs.org/docs/deploying-and-hosting/

**Comment by muescha** [comment]
also in the sidebar?

**Comment by muescha** [comment]
i see some screenshots on your twitter:
- maybe add these also to the doc
- there more `web.config` there for different folders - maybe add them also?

File: docs/docs/deploying-to-iis.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Install the [IIS URL Rewrite](https://www.iis.net/downloads/microsoft/url-rewrite) module if it's not already there.
+It's required to define the `outboundRules` in the `web.config` given below.
+
+Add a `web.config` file to the `static` folder inside your Gatsby solution. It will be copied unchanged to the `public` folder
+during `gatsby build`.
+
+Please note that local settings you make for your web in IIS Manager change the `web.config` file;
+you have to be careful to copy any changes to `web.config` on your server back to the version in your Gatsby `static` folder.
+
+```xml
```
Comment (muescha) [inline docs/docs/deploying-to-iis.md]:
```suggestion
```xml:file=static\web.config
```

**Comment by muescha** [comment]
you should run prettifier over the md to make the inline xml more consistent intends (2spaces, 4spaces, tabs)

**Comment by HaraldMuehlhoffCC** [comment]
@muescha Thank you for your comments! Please bear with me as I'm not yet very experienced in GitHub contribution ... ;)  

https://www.gatsbyjs.org/docs/deploying-and-hosting/ is using `<GuideList slug={props.slug} />` to list the guides. Hopefully my link might be added automatically because of some naming convention but I'm unsure because I didn't find where the component gets it's argument from. Any help would be appreciated!

> maybe add the pic with the rewrite-rule also to the doc? maybe in english?

As the rewrite-rules are all in the `web.config` and no usage of the IIS manager is required an image showing the editor might confuse some users. Also I've integrated all three web.configs into one.

> you should run prettifier over the md to make the inline xml more consistent intends (2spaces, 4spaces, tabs) - i don't know the defaults

I've converted tabs into spaces to make the output more consistent.

**Comment by muescha** [comment]
ur welcome :)

> GuideList

oh sorry - yes the `GuideList` is driven by the sitebar yaml ( `/www/src/data/sidebars/doc-links.yaml` ) it displays all childs under the `/docs/deploying-and-hosting/` slug in this case:

https://github.com/gatsbyjs/gatsby/blob/e3df29a716481242e1b8f6d7ee014dec3846bd29/www/src/data/sidebars/doc-links.yaml#L31-L72

> Screenshot

ok - then it is ok - maybe for a short check for the user, but since there not all infos there are displayed it is not needed

> make the output more consistent

great.

File: docs/docs/deploying-to-iis.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        <system.webServer>
+            <httpProtocol>
+                <customHeaders>
+                    <remove name="cache-control" />
+                    <add name="cache-control" value="cache-control: public, max-age=31536000, immutable" />
+                </customHeaders>
+            </httpProtocol>
+        </system.webServer>
+    </location>
+
```
Comment (muescha) [inline docs/docs/deploying-to-iis.md]: prettier don't like this ...
```suggestion

```

File: docs/docs/deploying-to-iis.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Install the [IIS URL Rewrite](https://www.iis.net/downloads/microsoft/url-rewrite) module if it's not already there.
+It's required to define the `outboundRules` in the `web.config` given below.
+
+Add a `web.config` file to the `static` folder inside your Gatsby solution. It will be copied unchanged to the `public` folder
+during `gatsby build`.
+
+Please note that local settings you make for your web in IIS Manager change the `web.config` file;
+you have to be careful to copy any changes to `web.config` on your server back to the version in your Gatsby `static` folder.
+
+```xml:file=static\web.config
```
Comment (laurieontech) [inline docs/docs/deploying-to-iis.md]:
```suggestion
```xml:file=static/web.config
```

**Comment by laurieontech** [comment]
Thanks for the PR! This is looking really good.

File: docs/docs/deploying-to-iis.md (hunk 3 above)
Comment (muescha) [reply docs/docs/deploying-to-iis.md]: iis is only on Windows and there is the backslash for filepaths

Comment (laurieontech) [reply docs/docs/deploying-to-iis.md]: Noted. I wasn't sure if it was there for a reason, or just a minor error that hadn't been caught! Will resolve.

**Pull request merged**